src/control/actor/host/notify.go

```diff
@@ -59,39 +58,32 @@ func (c *HostNotifyCmd) listenCloseF(net network.Network, addr ma.Multiaddr) {
 
 func (c *HostNotifyCmd) connectedF(net network.Network, conn network.Conn) {
 	logrus.WithFields(logrus.Fields{
-		"EVENT": "Connection detected",
-		"DIRECTION": fmtDirection(conn.Stat().Direction),
+		"EVENT":     "Connection detected",
+		"DIRECTION": conn.Stat().Direction.String(),
 	}).Info("Peer: ", conn.RemotePeer().String())
-
 	// try to request metadata for the peer
 	peerData, err := PollPeerMetadata(conn.RemotePeer(), c.Base, c.PeerMetadataState, c.Store, c.PeerStore)
```

Comment:
be careful here, you are not really handling the `err`. In line 68 you print an error message, but you continue executing the program, which can lead to an unexpected behaviour. i.e. if there is an `err` most likely `peerData` is `nil` or empty, and you can have problems later on.

Reply:
The command from Rumor that we are using to ping the BeaconMetadata, always records on the Rumor.Peerstore the peer, even though if the request wasn't successful. 
In this case, peerData will be always a full PeerAllData struct, since we are reading it from the Rumor.Peerstore after it is recorded, the error here just shows if the metadata of the given peer was successful or not.
I would really like to implement the changes on this function from #21 , it makes everything easier to understand. However, I agree with you on merging that PR in a separate moment, since this PR is already taking so many lines to read

Reply:
all good then, if it will be addresses in #21. resolving